### PR TITLE
Add Ingress functionality

### PR DESCRIPTION
**Pull request by eldios**
Add Ingress functionality

### CODE REVIEW

**Comment by a-palchikov** [comment]
Looks like the app update hook is failing:
```
2020-04-23T02:16:21Z DEBU [APP]       Job "ingress-app-update-d72fa7" in namespace "kube-system" event:ADDED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Job "ingress-app-update-d72fa7" in namespace "kube-system" event:ADDED utils/logginghook.go:56
2020-04-23T02:16:21Z INFO             Created Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:21Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU             now: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:ADDED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Job "ingress-app-update-d72fa7" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Job "ingress-app-update-d72fa7" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       now: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:ADDED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:21Z INFO             Container "hook" created, current state is "waiting, reason PodInitializing". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:21Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       now: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:21Z DEBU [APP]       now: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:24Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:24Z DEBU [APP]       3 seconds elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:31Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:31Z DEBU [APP]       10 seconds elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:32Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:32Z INFO             Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system", has changed state from "Pending" to "Running". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:32Z INFO             Container "hook" changed status from "waiting, reason PodInitializing" to "running". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:32Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:16:32Z DEBU [APP]       Start streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:16:32Z DEBU [APP]       11 seconds elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:16:32Z INFO             Release "gravity-ingress" does not exist. Installing it now. phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:32Z INFO             INSTALL FAILED phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:32Z INFO             PURGING CHART phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:32Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:33Z INFO             Successfully purged a chart! phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:33Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:34Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:21:34Z INFO             Container "hook" changed status from "running" to "terminated, exit code 1". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:34Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:34Z DEBU [APP]       5 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:21:34Z DEBU [APP]       Copy finished: copied: 269, result: <nil>. utils/logginghook.go:56
2020-04-23T02:21:34Z DEBU [APP]       Stopped streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:21:35Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:21:35Z INFO             Container "hook" restarted, current state is "running". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:35Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:21:35Z DEBU [APP]       Start streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:21:35Z DEBU [APP]       5 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:21:35Z INFO             Release "gravity-ingress" does not exist. Installing it now. phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:34Z INFO             INSTALL FAILED phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:34Z INFO             PURGING CHART phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:34Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:35Z INFO             Successfully purged a chart! phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:35Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:36Z DEBU [APP]       Copy finished: copied: 269, result: <nil>. utils/logginghook.go:56
2020-04-23T02:26:36Z DEBU [APP]       Stopped streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:26:36Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:36Z INFO             Container "hook" changed status from "running" to "terminated, exit code 1". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:36Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:36Z DEBU [APP]       10 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:50Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:50Z INFO             Container "hook" changed status from "terminated, exit code 1" to "waiting, reason CrashLoopBackOff". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:50Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:50Z DEBU [APP]       10 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:51Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:51Z INFO             Container "hook" restarted, current state is "running". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:51Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:26:51Z DEBU [APP]       Start streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:26:51Z DEBU [APP]       10 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:26:51Z INFO             Release "gravity-ingress" does not exist. Installing it now. phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:51Z INFO             INSTALL FAILED phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:51Z INFO             PURGING CHART phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:51Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:51Z INFO             Successfully purged a chart! phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:51Z INFO             Error: release gravity-ingress failed: timed out waiting for the condition phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:52Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:31:52Z INFO             Container "hook" changed status from "running" to "terminated, exit code 1". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:52Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:31:52Z DEBU [APP]       15 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:31:52Z DEBU [APP]       Copy finished: copied: 269, result: <nil>. utils/logginghook.go:56
2020-04-23T02:31:52Z DEBU [APP]       Stopped streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:32:03Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:32:03Z INFO             Container "hook" changed status from "terminated, exit code 1" to "waiting, reason CrashLoopBackOff". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:32:03Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:32:03Z DEBU [APP]       15 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:32:19Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:32:19Z INFO             Container "hook" restarted, current state is "running". phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:32:19Z INFO             phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:32:19Z DEBU [APP]       Start streaming logs for "Pod \"ingress-app-update-d72fa7-5fkfp\" in namespace \"kube-system\"", container "hook". utils/logginghook.go:56
2020-04-23T02:32:19Z DEBU [APP]       15 minutes elapsed: job kube-system/ingress-app-update-d72fa7 not yet complete (succeeded: 0, active: 1) event:MODIFIED utils/logginghook.go:56
2020-04-23T02:32:19Z INFO             Release "gravity-ingress" does not exist. Installing it now. phase:/runtime/ingress-app utils/logginghook.go:56
2020-04-23T02:36:21Z DEBU [APP]       Pod "ingress-app-update-d72fa7-5fkfp" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:36:21Z DEBU [APP]       Job "ingress-app-update-d72fa7" in namespace "kube-system" event:MODIFIED utils/logginghook.go:56
2020-04-23T02:36:21Z DEBU [APP]       Failed: Job was active longer than specified deadline. event:MODIFIED utils/logginghook.go:56
2020-04-23T02:36:21Z WARN             All attempts failed. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError Job was active longer than specified deadline
Stack Trace:
	/gopath/src/github.com/gravitational/gravity/lib/app/hooks/hooks.go:376 github.com/gravitational/gravity/lib/app/hooks.(*Runner).evalJobStatus
	/gopath/src/github.com/gravitational/gravity/lib/app/hooks/hooks.go:236 github.com/gravitational/gravity/lib/app/hooks.(*Runner).Wait.func1
	/gopath/src/github.com/gravitational/gravity/lib/utils/retry.go:237 github.com/gravitational/gravity/lib/utils.RetryWithInterval.func1
	/gopath/src/github.com/gravitational/gravity/vendor/github.com/cenkalti/backoff/retry.go:37 github.com/gravitational/gravity/vendor/github.com/cenkalti/backoff.RetryNotify
	/gopath/src/github.com/gravitational/gravity/lib/utils/retry.go:236 github.com/gravitational/gravity/lib/utils.RetryWithInterval
	/gopath/src/github.com/gravitational/gravity/lib/app/hooks/hooks.go:231 github.com/gravitational/gravity/lib/app/hooks.(*Runner).Wait
	/gopath/src/github.com/gravitational/gravity/lib/app/apps.go:113 github.com/gravitational/gravity/lib/app.WaitAppHook
	/gopath/src/github.com/gravitational/gravity/lib/app/service/app.go:361 github.com/gravitational/gravity/lib/app/service.(*applications).WaitAppHook
	/gopath/src/github.com/gravitational/gravity/lib/app/apps.go:65 github.com/gravitational/gravity/lib/app.StreamAppHook.func2
	/gopath/src/github.com/gravitational/gravity/lib/utils/retry.go:91 github.com/gravitational/gravity/lib/utils.Retry
	/gopath/src/github.com/gravitational/gravity/lib/app/apps.go:64 github.com/gravitational/gravity/lib/app.StreamAppHook
	/gopath/src/github.com/gravitational/gravity/lib/update/cluster/phases/app.go:223 github.com/gravitational/gravity/lib/update/cluster/phases.(*phaseApp).runHooks
	/gopath/src/github.com/gravitational/gravity/lib/update/cluster/phases/app.go:86 github.com/gravitational/gravity/lib/update/cluster/phases.(*updatePhaseApp).Execute
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:454 github.com/gravitational/gravity/lib/fsm.(*FSM).executeOnePhase
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:386 github.com/gravitational/gravity/lib/fsm.(*FSM).executePhaseLocally
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:346 github.com/gravitational/gravity/lib/fsm.(*FSM).executePhase
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:207 github.com/gravitational/gravity/lib/fsm.(*FSM).ExecutePhase
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:397 github.com/gravitational/gravity/lib/fsm.(*FSM).executeSubphasesSequentially
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:391 github.com/gravitational/gravity/lib/fsm.(*FSM).executePhaseLocally
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:318 github.com/gravitational/gravity/lib/fsm.(*FSM).executePhase
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:207 github.com/gravitational/gravity/lib/fsm.(*FSM).ExecutePhase
	/gopath/src/github.com/gravitational/gravity/lib/fsm/fsm.go:164 github.com/gravitational/gravity/lib/fsm.(*FSM).ExecutePlan
	/gopath/src/github.com/gravitational/gravity/lib/update/updater.go:173 github.com/gravitational/gravity/lib/update.(*Updater).executePlan
	/gopath/src/github.com/gravitational/gravity/lib/update/updater.go:61 github.com/gravitational/gravity/lib/update.(*Updater).Run.func1
	/go/src/runtime/asm_amd64.s:1337 runtime.goexit

```

**Comment by r0mant** [comment]
@eldios So I played with this a little bit and here's what I found. First, a couple of smaller notes.

We currently install this to kube-system namespace. I wonder if we should install it into its own namespace? E.g. `nginx-ingress`, similar to how OpenEBS stuff goes into `openebs`.

```
ubuntu@node-1:~/installer$ kubectl get pods -A | grep nginx
kube-system   gravity-ingress-nginx-ingress-controller-sxhgp                   1/1     Running     0          14m
kube-system   gravity-ingress-nginx-ingress-default-backend-69945449cb-xjc8x   1/1     Running     0          14m
```

Next, the nginx controller binds to a whole bunch of ports:

```
ubuntu@node-1:~/installer$ sudo netstat -lpten | grep nginx
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      101        9763947    32574/nginx: master
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      101        9763939    32574/nginx: master
tcp        0      0 127.0.0.1:10245         0.0.0.0:*               LISTEN      101        9761605    32455/nginx-ingress
tcp        0      0 127.0.0.1:10246         0.0.0.0:*               LISTEN      101        9763943    32574/nginx: master
tcp        0      0 127.0.0.1:10247         0.0.0.0:*               LISTEN      101        9763944    32574/nginx: master
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      101        9763945    32574/nginx: master
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      101        9763937    32574/nginx: master
tcp        0      0 0.0.0.0:8181            0.0.0.0:*               LISTEN      101        9763949    32574/nginx: master
tcp        0      0 0.0.0.0:8181            0.0.0.0:*               LISTEN      101        9763941    32574/nginx: master
```

I understand 80/443 but what is all that other stuff? Do we need it? Can we disable it?

Finally, upgrade indeed doesn't work. It's very easy to reproduce, just upgrade from 7.0 to this branch. Looking into why, I think the reason is the `--atomic` flag that says:

> If set, upgrade process rolls back changes made in case of failed upgrade, also sets --wait flag

Note the last part, about --wait flag:

> If set, will wait until all Pods, PVCs, Services, and minimum number of Pods of a Deployment are in a ready state before marking the release as successful. It will wait for as long as --timeout

Looking at the created objects, everything is Ready except for the controller Service which is of type LoadBalancer and stays Pending which I believe is why Helm blocks. Just testing this with plain Helm, if I provide `--set controller.service.type=NodePort` flag, it can install the release successfully.

I think hooks can be smarter - there's an environment variable available in every hook, `GRAVITY_CLUSTER_PROVIDER`, that contains the provider so hooks can set appropriate service type depending on which provider is used (i.e. LoadBalancer for "aws" or "gce" and NodePort for "generic" and everything else).

**Comment by eldios** [comment]
> We currently install this to kube-system namespace. I wonder if we should install it into its own namespace? E.g. `nginx-ingress`, similar to how OpenEBS stuff goes into `openebs`.

Done :+1: 

> Next, the nginx controller binds to a whole bunch of ports:
> 
> ```
> ubuntu@node-1:~/installer$ sudo netstat -lpten | grep nginx
> tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      101        9763947    32574/nginx: master
> tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      101        9763939    32574/nginx: master
> tcp        0      0 127.0.0.1:10245         0.0.0.0:*               LISTEN      101        9761605    32455/nginx-ingress
> tcp        0      0 127.0.0.1:10246         0.0.0.0:*               LISTEN      101        9763943    32574/nginx: master
> tcp        0      0 127.0.0.1:10247         0.0.0.0:*               LISTEN      101        9763944    32574/nginx: master
> tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      101        9763945    32574/nginx: master
> tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      101        9763937    32574/nginx: master
> tcp        0      0 0.0.0.0:8181            0.0.0.0:*               LISTEN      101        9763949    32574/nginx: master
> tcp        0      0 0.0.0.0:8181            0.0.0.0:*               LISTEN      101        9763941    32574/nginx: master
> ```
> 
> I understand 80/443 but what is all that other stuff? Do we need it? Can we disable it?

Unfortunately it seems that most of those are internal ports opened by the nginx ingress controller:

eg:
```
--
        # backend for when default-backend-service is not configured or it does not have endpoints
        server {
                listen 8181 default_server reuseport backlog=511;
                listen [::]:8181 default_server reuseport backlog=511;
--
        # default server, used for NGINX healthcheck and access to nginx stats
        server {
                listen 127.0.0.1:10246;
```

> Looking at the created objects, everything is Ready except for the controller Service which is of type LoadBalancer and stays Pending which I believe is why Helm blocks. Just testing this with plain Helm, if I provide `--set controller.service.type=NodePort` flag, it can install the release successfully.

Done :+1: 

> I think hooks can be smarter - there's an environment variable available in every hook, `GRAVITY_CLUSTER_PROVIDER`, that contains the provider so hooks can set appropriate service type depending on which provider is used (i.e. LoadBalancer for "aws" or "gce" and NodePort for "generic" and everything else).

This is the only remaining part ... working on it :desktop_computer:

**Comment by eldios** [comment]
@r0mant the final parti about the Clout Provider has also been added.. tests are passing too, so could you please review and approve this, when you get a chance?

Thanks!

**Comment by r0mant** [comment]
@eldios I looked at the changes in ingress-app and left a couple of comments: https://github.com/gravitational/ingress-app/pull/1#pullrequestreview-407771715. I think once those are addressed, we should be good to merge everything.

Also, don't forget to replace `0.0.3-dev` tag with something non "dev".

Thanks!

File: build.assets/Makefile [hunk 1]
```diff
@@ -28,6 +28,8 @@ LOGGING_APP_TAG ?= 0.0.1
 LOGGING_APP_BRANCH ?= $(LOGGING_APP_TAG)
 MONITORING_APP_TAG ?= 0.0.1
 MONITORING_APP_BRANCH ?= $(MONITORING_APP_TAG)
+INGRESS_APP_TAG ?= 0.0.1-dev
```
Comment (r0mant) [inline build.assets/Makefile:31]: Still says "-dev" here.

**Review by r0mant — APPROVED** [review]

File: build.assets/Makefile (hunk 1 above)
Comment (eldios) [reply build.assets/Makefile]: why are we duplicating those TAGs? that seems dangerous and error prone. 
Anyway fixed it now :+1:

**Pull request merged**